src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/CreateNewImageTests.cs

```diff
@@ -72,6 +72,79 @@ public void CreateNewImage_Baseline()
         newProjectDir.Delete(true);
     }
 
+    [DockerAvailableFact]
+    public void CanParseAspnetCoreUrls()
+    {
+        DirectoryInfo newProjectDir = new(GetTestDirectoryName());
+        if (newProjectDir.Exists)
+        {
+            newProjectDir.Delete(recursive: true);
+        }
+
+        newProjectDir.Create();
+
+        new DotnetNewCommand(_testOutput, "console", "-f", ToolsetInfo.CurrentTargetFramework)
+            .WithVirtualHive()
+            .WithWorkingDirectory(newProjectDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        new DotnetCommand(_testOutput, "publish", "-c", "Release", "-r", "linux-arm64", "--no-self-contained")
+            .WithWorkingDirectory(newProjectDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        CreateNewImage task = new();
+
+        (IBuildEngine buildEngine, List<string?> errors) = SetupBuildEngine();
+        task.BuildEngine = buildEngine;
+
+        task.BaseRegistry = "mcr.microsoft.com";
+        task.BaseImageName = "dotnet/runtime";
+        task.BaseImageTag = "8.0-preview";
+
+        // set up a bunch of variations on the aspnetcore urls
+        task.ContainerEnvironmentVariables = new[] {
+            new TaskItem("ASPNETCORE_URLS", new Dictionary<string, string> { { "Value", "https://*:12345;http://+:1234;http://localhost:123;http://1.2.3.4:12" } }) ,
+            new TaskItem("ASPNETCORE_HTTP_PORTS", new Dictionary<string, string> { { "Value", "999;666" } }),
+            new TaskItem("ASPNETCORE_HTTPS_PORTS", new Dictionary<string, string> { { "Value", "456;789" } })
+        };
```

Comment:
consider moving those as datasets instead to have more granular failure in case the logic is broken (however it might be worse test run time wise).

Reply:
Moving the logic to ImageBuilder will make it easier to verify this, so moving to datasets wouldn't turn into a huge performance hit. I'll dig more into that :+1:

Reply:
I was able to do this and added to the ImageBuilderTests in the UnitTest project